packages/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -610,6 +625,8 @@ export const useGridColumnResize = (
       const doc = ownerDocument(apiRef.current.rootElementRef!.current);
       doc.body.style.cursor = 'col-resize';
 
+      previousMouseClickEvent.current = event;
+
```

Comment:
I seem to recall something about react reusing its events, not sure if it still applies, but maybe we could store the native event directly?

Reply:
I stored it just for the comparison when mouseUp event occur. 

And used ref to persist it between renders

Reply:
yes you are correct I just checked we need to store the nativeEvent instead of syntheticEvent